internal/cmd/beta/volume/update/update.go

```diff
@@ -0,0 +1,173 @@
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			volumeLabel, err := iaasUtils.GetVolumeName(ctx, apiClient, model.ProjectId, model.VolumeId)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get volume name: %v", err)
+				volumeLabel = model.VolumeId
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to update volume %q?", volumeLabel)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("update volume: %w", err)
+			}
+
+			return outputResult(p, model, volumeLabel, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().StringP(nameFlag, "n", "", "Volume name")
+	cmd.Flags().String(descriptionFlag, "", "Volume description")
+	cmd.Flags().StringToString(labelFlag, nil, "Labels are key-value string pairs which can be attached to a volume. A label can be provided with the format key=value and the flag can be used multiple times to provide a list of labels")
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	volumeId := inputArgs[0]
+
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		Name:            flags.FlagToStringPointer(p, cmd, nameFlag),
+		VolumeId:        volumeId,
+		Description:     flags.FlagToStringPointer(p, cmd, descriptionFlag),
+		Labels:          flags.FlagToStringToStringPointer(p, cmd, labelFlag),
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiUpdateVolumeRequest {
+	req := apiClient.UpdateVolume(ctx, model.ProjectId, model.VolumeId)
+
+	var labelsMap *map[string]interface{}
+	if model.Labels != nil && len(*model.Labels) > 0 {
+		// convert map[string]string to map[string]interface{}
+		labelsMap = utils.Ptr(map[string]interface{}{})
+		for k, v := range *model.Labels {
+			(*labelsMap)[k] = v
+		}
+	}
+
+	payload := iaas.UpdateVolumePayload{
+		Name:        model.Name,
+		Description: model.Description,
+		Labels:      labelsMap,
+	}
+
+	return req.UpdateVolumePayload(payload)
+}
+
+func outputResult(p *print.Printer, model *inputModel, projectLabel string, volume *iaas.Volume) error {
```

Comment:
From the `RunE`, we are passing the volume name as third argument (and not the project label)

Reply:
The output text in `default:` needs to be adjusted also